### DISCUSSION TITLE
What are the recommendations to initialize dependencies

### DISCUSSION THREAD

**Discussion by paulomorgado** [microsoft/aspire#1809] [Q&A] [answered]
In my Aspire project I have service dependencies like databases, queues, etc.

What are the recommended ways to initialize/seed these dependencies before the services start?

**Comment by engineering87** [comment]
Hi @paulomorgado,
in the Aspire app, the key point is that your services should only start after dependent resources (databases, queues, caches) are initialized.
You can achieve this by adding a small initializer project that runs first, or by using `WaitFor` to enforce startup order.

```csharp
var db = builder.AddPostgres("pg")
    .WithImage("postgres:16");

var init = builder.AddProject<Projects.DbInitializer>("db-init")
    .WithReference(db) 
    .WaitFor(db);         // waits until DB container is ready

builder.AddProject<Projects.MyService>("my-service")
    .WithReference(db)
    .WaitFor(init);       // starts only after init finishes
```

This way, dependent services only start once the resources are ready in healthy state.

**Reply by davidfowl** [reply, marked as answer]
You can also handle OnResourceReady and put it in there if this is for development only.
